### PR TITLE
Add Readme.me and aggreation scripts

### PR DESCRIPTION
**Pull request by WojciechMazur**

### CODE REVIEW

File: Scala/README.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+- `mem` - perform single run of benchmark and collect measurments using /usr/bin/time eg. peak memory usage, however for Scala it does write outputs to defined file in csv format instead printing to stdout. 
+
+Additionally, we define additional tasks that are specific to Scala:
+- `clean` - remove all temporal files (*.tasty, *.class)
+- `test` - run benchmark with predefined output and check if it produces correct output
+- `measureWithWarmup` - use sRAPL library to run benchmarks using single JVM instance (measure starts new instance for each iteration)
+
+# Makefile configuration
+Makefile contains settings that can be used when running them:
+- `configName` - Name of the config to use, results with written to file with name derived from this value.
```
Comment (julienrf) [inline Scala/README.md]:
```suggestion
- `configName` - Name of the config to use, results will be written to file with name derived from this value.
```

File: Scala/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+- `output` - pass output redirect options, eg `output=>/dev/null` to ignore output as it logging to console can significetlly slow down execution
+- `mode` - use non standard benchmark implementation (applies only to `compile`), supported modes: 
+  - `mode=idiomatic` - Use Scala idiomatic implementation
+- `scalaPath` - path to `scala` binary
+- `scalacPath` - path to `scalac` binary
+
+# Makefiles management
+In general Makefiles should not be edited manually, instead `genMakefile.scala` script should be used. This ensures that they would stay consistent thanks to usage of common template. 
+Recommended way of running `genMakefile.scala` is via [scala-cli](https://scala-cli.virtuslab.org/) eg. `scala-cli genMakefile.scala` - running this command would update Makefiles in all benchmark directories. Alternativelly it can be compiled and runned manually using `scalac` and `scala`/`java` programs. 
+`genMakefile.scala` needs Scala in version 3.0.0 or higher.
```
Comment (julienrf) [inline Scala/README.md:36]: scala-cli will automatically use the right version of Scala.

File: Scala/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  cd EnergyLanugages/Scala 
+  #make -C <benchmark directory> <commands*>
+  make -C binary-trees clean compile test measure measureWithWarmup
+  # alternatively you can run it from inside of benchmark directory
+  cd binary-trees
+  make clean compile test measure measureWithWarmup 
+  # run idiomatic implementation and set change config name
+  make clean compile test measure measureWithWarmup mode=idiomatic configName=Scala-idiomatic
+  ```
+It is recommended to redirect standard output from benchmarks as writing to the console can slow down execution
```
Comment (julienrf) [inline Scala/README.md:60]: When you ran the benchmarks, did you redirect the output for all the languages as well?

File: Scala/README.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+# Makefile configuration
+Makefile contains settings that can be used when running them:
+- `configName` - Name of the config to use, results with written to file with name derived from this value.
+- `warmup-iterations` - number of initial warmup iterations (used only in `measureWithWarmup`)
+- `warmup-measure` - should collect measurments when performing warmup iterations, results would be logged using `<benchmark-name>_warmup` label in the results (used only in `measureWithWarmup`)
+- `output` - pass output redirect options, eg `output=>/dev/null` to ignore output as it logging to console can significetlly slow down execution
+- `mode` - use non standard benchmark implementation (applies only to `compile`), supported modes: 
+  - `mode=idiomatic` - Use Scala idiomatic implementation
+- `scalaPath` - path to `scala` binary
```
Comment (julienrf) [inline Scala/README.md]: Maybe add
```suggestion
- `scalaPath` - path to `scala` binary. Default value assumes that `scala` has been installed via SDKMAN.
```

File: Scala/README.md (hunk 2 above)
Comment (WojciechMazur) [reply Scala/README.md:36]: Yes, however, if you don't use scala-cli, but manually `scalac` and running it using `scala` or `java`  you need to make sure that the used version is correct, that's why I've added this information.

File: Scala/README.md (hunk 3 above)
Comment (WojciechMazur) [reply Scala/README.md:60]: When running benchmarks I've used `compile_all.py` which does always redirect output when running benchmarks.

**Pull request merged**